### PR TITLE
Fix/banner not opening

### PR DESCRIPTION
**Pull request by jdahlke**
Pivotal: 

[Cookie Consent dialog does not pop up, if consent not set of www.bundesimmobilien.de](https://www.pivotaltracker.com/story/show/184081546)

* and added tests

```shell
% npm run test

 PASS  src/Components/CookieBanner.test.js
  <CookieBanner />
    in simple mode
      ✓ shows header and 3 buttons (277 ms)
    in extended mode
      ✓ shows header and 3 buttons (121 ms)
    when consents in cookie are set
      ✓ does not show modal (13 ms)
```

* and added debug output
```
window.localStorage.setItem("ScrivitoCookieBanner.debug", "true")`
```

### CODE REVIEW

File: babel.config.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module.exports = function (api) {
+  api.cache(true)
+
+  const presets = ['@babel/preset-env', '@babel/preset-react']
+  const plugins = [
+    // '@babel/plugin-proposal-class-properties',
+    // '@babel/plugin-proposal-export-default-from',
+    // '@babel/plugin-proposal-export-namespace-from',
+    // '@babel/plugin-proposal-object-rest-spread'
```
Comment (natalka) [inline babel.config.js]: do we need it?

File: src/Components/CookieBanner.js [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
 import { useCookieConsent } from "./CookieConsentContext";
 import Header from "./Header";
 import Body from "./Body";
 import Footer from "./Footer";
 import "../assets/main.scss";
 
 export default function CookieBanner() {
-  const { bannerVisibility, switchBannerMode, I18n } = useCookieConsent();
+  const { bannerVisibility, switchBannerMode, I18n, config } =
+    useCookieConsent();
```
Comment (natalka) [inline src/Components/CookieBanner.js:12]: Why would you:
1) you grab config
`const { bannerVisibility, switchBannerMode, I18n, config } =
    useCookieConsent();`
 
2) Pass config into `<Body config={config} />`

3) and than in Body:
```
export default function Body({ config }) {
  const { isExtendedMode, I18n } = useCookieConsent();
```
?
  
  It's confusing. Why don't we pass all params (or something)?

File: src/Components/CookieConsentContext.js [hunk 3]
```diff
@@ -44,54 +45,41 @@ export function CookieConsentProvider({
...
-      const cookieValue = Cookies.get(consentCookieName);
-      try {
-        return JSON.parse(cookieValue);
-      } catch (e) {
-        return {};
-      }
-    };
+    const value = getCookieValue(consentCookieName);
+
+    logger.debug("[ScrivitoCookieBanner] initialize with:", value);
```
Comment (natalka) [inline src/Components/CookieConsentContext.js:50]: never again in that rabbit hole 👍 🤣

File: src/Components/CookieConsentContext.js [hunk 4]
```diff
@@ -111,9 +99,11 @@ export function CookieConsentProvider({
...
     Object.keys(cookieConsentChoice).length === 0;
 
-  const saveAndClose = () => {
-    if (isCookieConsentEmpty()) {
-      setDecisionForAllCookies(DECLINED);
+  const saveAndClose = (decision) => {
+    if (decision === ACCEPTED || decision === DECLINED) {
+      setCookieDecision(editableCookies, decision);
+    } else if (isCookieConsentEmpty()) {
+      setCookieDecision(editableCookies, DECLINED);
```
Comment (natalka) [inline src/Components/CookieConsentContext.js:106]: was this an issue?

File: src/utils/defs.js [hunk 5]
```diff
@@ -0,0 +1,2 @@
+export const ACCEPTED = "accepted";
+export const DECLINED = "declined";
```
Comment (natalka) [inline src/utils/defs.js]: you called I `defs`, as the one who despises abbreviations 🤣 ? ok I would expand it cuz even I didn't got right away, or maybe name it `const`.

File: src/utils/logger.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
+/* eslint no-console: "off" */
+
+function info(...args) {
+  if (typeof console !== "undefined") {
```
Comment (natalka) [inline src/utils/logger.js]: just silly me wanna understand - why console would be undefined 🙀 ?

File: webpack.config.js [hunk 7]
```diff
@@ -43,7 +43,7 @@ module.exports = {
         test: /\.s?css$/,
         use: [
           { loader: MiniCssExtractPlugin.loader },
-          { loader: 'css-loader' }, 
+          { loader: 'css-loader' },
           { loader: 'sass-loader' },
```
Comment (natalka) [inline webpack.config.js:47]: hmm that one too 🤔 why css if we have sass?

File: babel.config.js (hunk 1 above)
Comment (jdahlke) [reply babel.config.js]: I have to check in details, but my quick thoughts would be

```
    // '@babel/plugin-proposal-class-properties', => no
    // '@babel/plugin-proposal-export-default-from', => 
    // '@babel/plugin-proposal-export-namespace-from', => no
    // '@babel/plugin-proposal-object-rest-spread' => yes
```

File: src/Components/CookieConsentContext.js (hunk 4 above)
Comment (jdahlke) [reply src/Components/CookieConsentContext.js:106]: No, this was just a little refactoring. I just thought that 3 function that do the same are not necessary.

File: src/utils/defs.js (hunk 5 above)
Comment (jdahlke) [reply src/utils/defs.js]: `src/utils/constants.js` or `src/utils/definitions.js`?

I don't really like `const` or `constants`, but I could live with it.

File: src/utils/logger.js (hunk 6 above)
Comment (jdahlke) [reply src/utils/logger.js]: Some browsers, though only IE comes to mind, didn't always support console.
But you are correct, it seems to be supported everywhere nowadays.

https://developer.mozilla.org/en-US/docs/Web/API/console

File: webpack.config.js (hunk 7 above)
Comment (jdahlke) [reply webpack.config.js:47]: No idea, have to check. I just removed some trailing whitespaces here. 🤷‍♂️

File: src/Components/CookieConsentContext.js [hunk 8]
```diff
@@ -44,54 +45,41 @@ export function CookieConsentProvider({
...
-    };
+    const value = getCookieValue(consentCookieName);
+
+    logger.debug("[ScrivitoCookieBanner] initialize with:", value);
+    setCookieConsentChoice(value);
 
-    setCookieConsentChoice(readCookieStore());
+    if (isEmpty(value)) {
+      setBannerVisibility(true);
+    }
```
Comment (jdahlke) [inline src/Components/CookieConsentContext.js:55]: @natalka That was the actual bug. Display banner, when cookie is missing.

It was removed by this commit and I assume this was by acciddent.
https://github.com/infopark-customers/scrivito-another-cookie-banner/commit/99f0c9cc2a1047143c85ca6740e3b5b990a7e707

File: src/utils/defs.js (hunk 5 above)
Comment (natalka) [reply src/utils/defs.js]: `src/utils/definitions.js` 👍

File: webpack.config.js (hunk 7 above)
Comment (jdahlke) [reply webpack.config.js:47]: It seems as if `css-loader` is always needed, when you use `sass-loader` too.

```
ERROR in ./src/assets/main.scss
Module build failed (from ./node_modules/mini-css-extract-plugin/dist/loader.js):
HookWebpackError: Module parse failed: Unexpected token (1:1)
File was processed with these loaders:
 * ./node_modules/sass-loader/dist/cjs.js
You may need an additional loader to handle the result of these loaders.
> ﻿.cookie-banner-iso{font-size:16px;/*!
```



> Chain the sass-loader with the [css-loader](https://github.com/webpack-contrib/css-loader) and the [style-loader](https://github.com/webpack-contrib/style-loader) to immediately apply all styles to the DOM or the [mini-css-extract-plugin](https://github.com/webpack-contrib/mini-css-extract-plugin) to extract it into a separate file.

Source: https://www.npmjs.com/package/sass-loader

**Review by natalka — APPROVED** [review]

**Pull request merged**